test/config/config.test.ts

```diff
@@ -125,34 +157,166 @@ describe('Config', () => {
+        return Promise.resolve(undefined)
+      }
+      }
+      const load = async (): Promise<void>  => {}
+      const findCommand = async (): Promise<ICommand.Class> => {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+      // @ts-ignore
+        return new MyComandClass()
+      }
+      const commandPluginA: ICommand.Plugin = {
+        strict: false,
+        aliases: [], args: [], flags: {}, hidden: false, id: commandIds[0], async load(): Promise<ICommand.Class> {
+          return new MyComandClass() as unknown as ICommand.Class
+        },
+        pluginType: types[0] ?? 'core',
+        pluginAlias: '@My/plugina',
+      }
+      const commandPluginB: ICommand.Plugin = {
+        strict: false,
+        aliases: [], args: [], flags: {}, hidden: false, id: commandIds[1], async load(): Promise<ICommand.Class> {
+          return new MyComandClass() as unknown as ICommand.Class
+        },
+        pluginType: types[1] ?? 'core',
+        pluginAlias: '@My/pluginb',
+      }
+      const hooks = {}
+      const pluginA: IPlugin = {load,
+        findCommand,
+        name: '@My/plugina',
+        alias: '@My/plugina',
+        commands: [commandPluginA],
+        _base: '',
+        pjson,
+        commandIDs: [commandIds[0]] as string[],
+        root: '',
+        version: '0.0.0',
+        type: types[0] ?? 'core',
+        hooks,
+        topics: [],
+        valid: true,
+        tag: 'tag',
+      }
+      commandPluginA
+      const pluginB: IPlugin = {
+        load,
+        findCommand,
+        name: '@My/pluginb',
+        alias: '@My/pluginb',
+        commands: [commandPluginB],
+        _base: '',
+        pjson,
+        commandIDs: [commandIds[1]] as string[],
+        root: '',
+        version: '0.0.0',
+        type: types[1] ?? 'core',
+        hooks,
+        topics: [],
+        valid: true,
+        tag: 'tag',
+      }
+      const plugins: IPlugin[] = [pluginA, pluginB]
+      let test = fancy
+      .resetConfig()
+      .env(env, {clear: true})
+      .stub(os, 'homedir', () => path.join(homedir))
+      .stub(os, 'platform', () => platform)
+
+      if (pjson) test = test.stub(util, 'loadJSON', () => Promise.resolve(pjson))
+      test = test.add('config', async () => {
+        const config = await Config.load()
+        config.plugins = plugins
+        config.pjson.oclif.plugins = ['@My/pluginb', '@My/plugina']
+        config.pjson.dependencies = {'@My/pluginb': '0.0.0', '@My/plugina': '0.0.0'}
+        return config
+      })
+      // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+      // @ts-ignore
+      return {
+        it(expectation: string, fn: (config: Interfaces.Config) => any) {
+          test
+          .do(({config}) => fn(config))
+          .it(expectation)
+          return this
+        },
+      }
+    }
+
+    findCommandTestConfig()
+    .it('find command with no duplicates', config => {
+      expect(config.findCommand('foo:bar', {must: true}))
```

Comment:
Test should make sure that the command is coming from the right plugin.
```suggestion
      const command = config.findCommand('foo:bar', {must: true})
      expect(command).to.have.property('pluginAlias', '@My/plugina')
```